transport-classes-io_uring/src/main/java/io/netty/channel/uring/AbstractIoUringChannel.java

```diff
@@ -1092,36 +1097,43 @@ public void connect(
+                    ByteBuf initialData = null;
+                    if (IoUring.isTcpFastOpenClientSideAvailable() &&
                         config().getOption(ChannelOption.TCP_FASTOPEN_CONNECT) == Boolean.TRUE) {
-                    ChannelOutboundBuffer outbound = unsafe().outboundBuffer();
-                    outbound.addFlush();
-                    Object curr;
-                    if ((curr = outbound.current()) instanceof ByteBuf) {
-                        initialData = (ByteBuf) curr;
+                        ChannelOutboundBuffer outbound = unsafe().outboundBuffer();
+                        outbound.addFlush();
+                        Object curr;
+                        if ((curr = outbound.current()) instanceof ByteBuf) {
+                            initialData = (ByteBuf) curr;
+                        }
                     }
-                }
-                if (initialData != null) {
-                    msgHdrMemoryArray = new MsgHdrMemoryArray((short) 1);
-                    MsgHdrMemory hdr = msgHdrMemoryArray.hdr(0);
-                    hdr.set(socket, inetSocketAddress, IoUring.memoryAddress(initialData),
-                            initialData.readableBytes(), (short) 0);
-
-                    int fd = fd().intValue();
-                    IoRegistration registration = registration();
-                    IoUringIoOps ops = IoUringIoOps.newSendmsg(fd, (byte) 0, Native.MSG_FASTOPEN,
-                            hdr.address(), hdr.idx());
-                    connectId = registration.submit(ops);
-                    if (connectId == 0) {
-                        // Directly release the memory if submitting failed.
-                        freeMsgHdrArray();
+                    if (initialData != null) {
+                        msgHdrMemoryArray = new MsgHdrMemoryArray((short) 1);
+                        MsgHdrMemory hdr = msgHdrMemoryArray.hdr(0);
+                        hdr.set(socket, inetSocketAddress, IoUring.memoryAddress(initialData),
+                                initialData.readableBytes(), (short) 0);
+
+                        int fd = fd().intValue();
+                        IoRegistration registration = registration();
+                        IoUringIoOps ops = IoUringIoOps.newSendmsg(fd, (byte) 0, Native.MSG_FASTOPEN,
+                                hdr.address(), hdr.idx());
+                        connectId = registration.submit(ops);
+                        if (connectId == 0) {
+                            // Directly release the memory if submitting failed.
+                            freeMsgHdrArray();
+                        }
+                    } else {
+                        submitConnect(inetSocketAddress);
                     }
-                } else {
-                    submitConnect(inetSocketAddress);
                 }
+
+                if (remoteAddress instanceof DomainSocketAddress) {
```

Comment:
Consider making this an `else if` to connect with the `remoteAddress instanceof InetSocketAddress` above, and then add an else-clause that throws if the remote address is of neither type.

If I read the code right, I think we previously just always threw a `ClassCastException`.

Reply:
done!
```java
else if (remoteAddress instanceof DomainSocketAddress) {
                    DomainSocketAddress unixDomainSocketAddress = (DomainSocketAddress) remoteAddress;
                    submitConnect(unixDomainSocketAddress);
                } else {
                    throw new Error("Unexpected SocketAddress implementation " + remoteAddress);
                }
```
Keep with socket::connect